compiler/src/iree/compiler/Dialect/VMVX/vmvx.imports.mlir

```diff
@@ -47,6 +46,315 @@ vm.import @add.2d.f32(
+
+  %out_buffer : !vm.buffer,
+  %out_offset : i64,
+  %out_strides : tuple<i64, i64>,
+
+  %sizes : tuple<i64, i64>
+)
+
+vm.import @muli.2d.x32(
+  %lhs_buffer : !vm.buffer,
+  %lhs_offset : i64,
+  %lhs_strides : tuple<i64, i64>,
+
+  %rhs_buffer : !vm.buffer,
+  %rhs_offset : i64,
+  %rhs_strides : tuple<i64, i64>,
+
+  %out_buffer : !vm.buffer,
+  %out_offset : i64,
+  %out_strides : tuple<i64, i64>,
+
+  %sizes : tuple<i64, i64>
+)
+
+vm.import @ori.2d.x32(
+  %lhs_buffer : !vm.buffer,
+  %lhs_offset : i64,
+  %lhs_strides : tuple<i64, i64>,
+
+  %rhs_buffer : !vm.buffer,
+  %rhs_offset : i64,
+  %rhs_strides : tuple<i64, i64>,
+
+  %out_buffer : !vm.buffer,
+  %out_offset : i64,
+  %out_strides : tuple<i64, i64>,
+
+  %sizes : tuple<i64, i64>
+)
+
+vm.import @shli.2d.x32(
+  %lhs_buffer : !vm.buffer,
+  %lhs_offset : i64,
+  %lhs_strides : tuple<i64, i64>,
+
+  %rhs_buffer : !vm.buffer,
+  %rhs_offset : i64,
+  %rhs_strides : tuple<i64, i64>,
+
+  %out_buffer : !vm.buffer,
+  %out_offset : i64,
+  %out_strides : tuple<i64, i64>,
+
+  %sizes : tuple<i64, i64>
+)
+
+vm.import @shrsi.2d.x32(
+  %lhs_buffer : !vm.buffer,
+  %lhs_offset : i64,
+  %lhs_strides : tuple<i64, i64>,
+
+  %rhs_buffer : !vm.buffer,
+  %rhs_offset : i64,
+  %rhs_strides : tuple<i64, i64>,
+
+  %out_buffer : !vm.buffer,
+  %out_offset : i64,
+  %out_strides : tuple<i64, i64>,
+
+  %sizes : tuple<i64, i64>
+)
+
+vm.import @shrui.2d.x32(
+  %lhs_buffer : !vm.buffer,
+  %lhs_offset : i64,
+  %lhs_strides : tuple<i64, i64>,
+
+  %rhs_buffer : !vm.buffer,
+  %rhs_offset : i64,
+  %rhs_strides : tuple<i64, i64>,
+
+  %out_buffer : !vm.buffer,
+  %out_offset : i64,
+  %out_strides : tuple<i64, i64>,
+
+  %sizes : tuple<i64, i64>
+)
+
+vm.import @subf.2d.x32(
+  %lhs_buffer : !vm.buffer,
+  %lhs_offset : i64,
+  %lhs_strides : tuple<i64, i64>,
+
+  %rhs_buffer : !vm.buffer,
+  %rhs_offset : i64,
+  %rhs_strides : tuple<i64, i64>,
+
+  %out_buffer : !vm.buffer,
+  %out_offset : i64,
+  %out_strides : tuple<i64, i64>,
+
+  %sizes : tuple<i64, i64>
+)
+
+vm.import @subi.2d.x32(
+  %lhs_buffer : !vm.buffer,
+  %lhs_offset : i64,
+  %lhs_strides : tuple<i64, i64>,
+
+  %rhs_buffer : !vm.buffer,
+  %rhs_offset : i64,
+  %rhs_strides : tuple<i64, i64>,
+
+  %out_buffer : !vm.buffer,
+  %out_offset : i64,
+  %out_strides : tuple<i64, i64>,
+
+  %sizes : tuple<i64, i64>
+)
+
+vm.import @xori.2d.x32(
+  %lhs_buffer : !vm.buffer,
+  %lhs_offset : i64,
+  %lhs_strides : tuple<i64, i64>,
+
+  %rhs_buffer : !vm.buffer,
+  %rhs_offset : i64,
+  %rhs_strides : tuple<i64, i64>,
+
+  %out_buffer : !vm.buffer,
+  %out_offset : i64,
+  %out_strides : tuple<i64, i64>,
+
+  %sizes : tuple<i64, i64>
+)
+
+//===----------------------------------------------------------------------===//
+// VMVX Unary Elementwise Kernels
+// Each is specialized by opcode, rank and type width.
+//===----------------------------------------------------------------------===//
+
+vm.import @absf.2d.x32(
```

Comment:
should these be `abs.2d.f32`, given that they only operate on f32? (they can all use the same shim internally, but it's kind of confusing to have both a typed op name and generic untyped suffix)

Reply:
Ah, I was just relying too much on some automated bits. Yes, I can change them to typed.

Reply:
they still have the redundant naming in them (vmvx.abs**f**.2d.**f**32) - I think you need to change the enum?

Reply:
Was debating that, but yeah... Ok, I can remove the 'f' and 'i' but still need to keep 's' and 'u' on some of them.